README.md

```diff
@@ -26,7 +26,7 @@ pip install intervaltree
 Features
 --------
 
-* Supports Python 2.7 and Python 3.5+ (Tested under 2.7, and 3.5 thru 3.8)
+* Supports Python 3.7+ (Tested under 3.7 thru 3.12)
```

Comment:
The wording here isn't exactly symmetrical to the old version where 3.9 was in development, but we are testing 3.12 so I thought this was Ok to say.